setup.py

```diff
@@ -11,16 +11,22 @@
 except ImportError:
     cythonize = None
 
-include_dirs = []
-try:
-    import numpy
 
-    include_dirs.append(numpy.get_include())
-except ImportError:
-    print("Numpy and its headers are required to run setup(). Exiting.")
-    sys.exit(1)
+class get_numpy_include:
+    """Helper class to determine the numpy include path
+    The purpose of this class is to postpone importing numpy
+    until it is actually installed, so that the ``get_include()``
+    method can be invoked. """
+
+    def __str__(self):
+        import numpy
+        return numpy.get_include()
```

Comment:
Maybe the `class` is overpriced! 
a simple function should be enough

```python
def get_numpy_include():
    import numpy
    yield numpy.get_include()
```

and the in the setup() do 
```python
    include_dirs=[get_numpy_include()],
    setup_requires=["numpy"],
```

🤷‍♂️

Reply:
`:homer:`

Reply:
![image](https://emoji.slack-edge.com/T01D2T6F8JD/homer/2a5f64dd1073adac.gif)